### DISCUSSION TITLE
Use icon from directory?

### DISCUSSION THREAD

**Discussion by konzertnr9** [Hofknecht/SystemTrayMenu#503] [Q&A] [answered]
What does "Use icon from directory" in Settings - Customize do exactly?
Is it about the individual shortcuts or the tray icon?

**Comment by topeterk** [comment]
Yes, it will take the icon from the root directory for the task bar and tray icons.

**Reply by konzertnr9** [reply]
Thank you.
What is the naming scheme for the icons?

**Reply by topeterk** [reply, marked as answer]
There is no scheme or I misunderstood the question.

You just use the usual exporer context menu to change/select the icon of your choice:
right click -> properties -> customize -> other symbol

So, the icon is taken from the directory entry which is usually stored in the Desktop.ini of the folder.

**Reply by konzertnr9** [reply]
Thank you for your clarification.
I thought it is a function to pick up icons from the directory based on their names (e.g. xxx.ico for xxx.exe) but I now understand that it is to respect the directory entry.

**Reply by JoergSieger** [reply]
Thank you very much for the great app. I have a problem under Windows 11. I want to use an icon of my own choosing. If I create a shortcut on the desktop, I can change the icon. However, the folder symbol is still always displayed in the taskbar and in the systray. What am I doing wrong?

**Reply by topeterk** [reply]
@JoergSieger maybe this helps: https://www.youtube.com/watch?v=Pbpe_KNEELM
You have to change the icon of the directory itself. It is not about any shortcut that leads to it.

**Reply by JoergSieger** [reply]
I expressed myself misleadingly. I created a shortcut to SystemtrayMenu on the desktop. I can change the icon here. However, this change is not reflected in the taskbar. The icon for a folder is simply displayed here. Likewise in the system tray itself.

**Reply by topeterk** [reply]
Oh yes, I see. This topic was about the directory icon and I thought your question was within that context.

As far as I know, the tray icon is either the default one or the one taken from the target root directory (as mentioned above).
I am not sure if that, what you are thinking of, is even possible. The application would have to know exactly from which shortcut it was started and I don't know of any way to figure that out.
However, if this is actually possible, it sounds like a nice idea, so let me know when someone knows how to do it, so we can think about integrating this feature.

**Reply by JoergSieger** [reply]
I thought of this because I saw a nice icon on various screenshots. For me (Win11) I only get the boring standard folder icon and unfortunately that can't be changed - or maybe it can?

**Reply by topeterk** [reply]
Have you changed the icon of the older as shown in the answer or the video above?
You should be able to change it to any available icon (you just have to have write permissions on the folder as the icon will be stored in a file called Desktop.ini within that folder). At this moment the icon for this folder should already change in Windows Explorer.
Then, when this folder ist set as root folder for STM, you checked the "Use icon from directory" setting and restarted STM once, than the icon will be used by STM aswell.

**Reply by JoergSieger** [reply]
Yes I changed it as shown. But unfortunately ist dosn't work.
![Image](https://github.com/Hofknecht/SystemTrayMenu/assets/164928969/e27e3d7a-bc0a-4fe9-b289-35a06b050b21)

**Reply by topeterk** [reply]
What you marked is a shortcut which has its own icon but that is not used by STM.
And I don't know how STM can figure out that it was started by this shortcut, so from my point of view there is no way that STM can find this shortcut and take its icon.

The only options that is implemented right know is the icon for the folder it self. So, this not about the shortcut (the *.lnk file, you marked).
Your marked file has a small icon on the button left shown with an arrow indicating that this is a shortcut. When you look at the video, you can see that this icon is missing. And as in this video, you have to do this with the actual folder.

When you right click on the STM icon in the task bar on the bottom left, as you have also marked in the screenshot. Then select "Settings". In the new dialog click on the "Open directory" button:
![image](https://github.com/Hofknecht/SystemTrayMenu/assets/4932525/bb62c558-ef11-4c96-87f9-3aaef4a13e7e)
This should open the root folder in an explorer window and this is the place where you have to change the icon for this folder.

**Reply by JoergSieger** [reply]
I understand. Now it works. Thanks a lot.
